internal/index/indexscanner/scanner_test.go

```diff
@@ -90,6 +93,56 @@ func TestReadPluginFile(t *testing.T) {
 	}
 }
 
+func TestReadReceiptFile(t *testing.T) {
```

Comment:
I'm not particularly happy with the test below, as it exercises only a few things. For example, it doesn't check whether the status fields are parsed correctly.

We just need 1 test case that makes sure the fields are parsed decently. and the ``json:"foo,omitempty"`` field-tags are accurate.

So I suggest we do this:

1. Add a few receipt test files in `testdata/receipts/foo.yaml`

2. Parse the receipt file from disk.

3. Build as `NewReceipt().WithXxx(...)....V()` and use `cmp.Diff` to do equality check.

@corneliusweig wdyt?

Reply:
Well, so far `Receipt` does not have any extra fields beside the ones from `Plugin`. So we are cutting corners by parsing a plugin yaml as a receipt. I think that's ok for now, but we need to ensure that we extend this test as suggested by Ahmet.

And I'm in for the receipt builder pattern. The sooner we have it, the better.

Reply:
Yeah in the next PR, we should add the `status` field and start working with it to develop tests like this (where we can roundtrip the type i.e. write/parse a YAML file).

Reply:
So is this ok as is for now while status doesn't exist?

Reply:
yeah not so useful but helps us pass the tests. 
feel free to remove it altogether or add TODO that captures comment above, but this should be the next PR anyway.

Reply:
Ok I'll just remove it from this PR and add the tests in the next one when they can be more meaningful with the new status field.